org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -209,6 +209,13 @@ public void didChangeConfiguration(DidChangeConfigurationParams params) {
 				unregisterCapability(Preferences.CODE_LENS_ID, Preferences.TEXT_DOCUMENT_CODE_LENS);
 			}
 		}
+		if (preferenceManager.getClientPreferences().isSignatureHelpDynamicRegistrationSupported()) {
+			if (preferenceManager.getPreferences().isSignatureHelpEnabled()) {
+				registerCapability(Preferences.SIGNATURE_HELP_ID, Preferences.TEXT_DOCUMENT_SIGNATURE_HELP, new CodeLensOptions(true));
```

Comment:
new SignatureHelpOptions(Arrays.asList("(")).

I suggest you had a `public static SignatureHelpOptions createOption(){...}` method to SignatureHelpHandler